src/loc/LocItem.tsx

```diff
@@ -123,112 +125,70 @@ export abstract class LocItem implements CommonData {
     }
 
-    abstract metadataData(): MetadataData;
-
-    abstract fileData(): FileData;
-
-    abstract linkData(): LinkData;
+    abstract title(): string;
 
-    abstract publish(timestamp: string | null, fees?: Fees, storageFeePaidBy?: string): LocItem;
+    hasData(): boolean {
+        return this.commonData.submitter !== undefined && this._data !== undefined;
+    }
 
-    get submitterOrThrow() {
-        if(this.commonData.submitter) {
-            return this.commonData.submitter;
+    data(): T {
+        if(this._data) {
+            return this._data;
         } else {
             throw new Error("Data not set");
         }
     }
-}
 
-export class MetadataItem extends LocItem {
-
-    constructor(commonData: CommonData, metadataData?: MetadataData) {
-        super(commonData);
-        this._metadataData = metadataData;
+    as<U>(): U {
+        return this._data as U;
     }
 
-    private _metadataData?: MetadataData;
+    abstract publish(timestamp: string | null, fees?: Fees, storageFeePaidBy?: string): AbstractLocItem<T>;
 
-    override title() {
-        return this._metadataData?.name || "-";
-    }
-
-    override hasData(): boolean {
-        return super.hasData() && this._metadataData !== undefined;
-    }
-
-    metadataData(): MetadataData {
-        if(this._metadataData) {
-            return this._metadataData;
+    get submitterOrThrow() {
+        if(this.commonData.submitter) {
+            return this.commonData.submitter;
         } else {
             throw new Error("Data not set");
         }
     }
+}
 
-    fileData(): FileData {
-        throw new Error("Method not implemented.");
-    }
+export class MetadataItem extends AbstractLocItem<MetadataData> {
 
-    linkData(): LinkData {
-        throw new Error("Method not implemented.");
+    static TYPE: LocItemType = "Data";
+    
+    override title() {
+        return this.data()?.name || "-";
     }
 
-    override publish(timestamp: string | null, fees?: Fees, _storageFeePaidBy?: string): LocItem {
+    override publish(timestamp: string | null, fees?: Fees, _storageFeePaidBy?: string): MetadataItem {
         return new MetadataItem(
             {
                 ...this.commonData,
                 timestamp,
                 fees,
             },
-            this._metadataData,
+            this.data(),
         );
     }
 }
 
-export class FileItem extends LocItem {
-
-    constructor(commonData: CommonData, fileData?: FileData) {
-        super(commonData);
-        this._fileData = fileData;
-    }
-
-    readonly _fileData?: FileData;
+export class FileItem extends AbstractLocItem<FileData> {
 
     override title() {
-        return this._fileData?.nature || "-";
-    }
-
-    override hasData(): boolean {
-        return super.hasData() && this._fileData !== undefined;
-    }
-
-    metadataData(): MetadataData {
-        throw new Error("Method not implemented.");
+        return this.data()?.nature || "-";
```

Comment:
this.data() does not return undefined or null so the ? is useless I think.

Reply:
Yse, indeed. `?` removed evrywhere it's useless.